### DISCUSSION TITLE
4 controller support for PS1 & PS2

### DISCUSSION THREAD

**Discussion by hpoom** [darthcloud/BlueRetro#465] [Ideas]
I note the wiring for PS2 only supports 2 controllers. I would like to be able to buy a PS2 multitap and mount an ESP32 inside to make an internal 4 controller blueretro for PS2.

Will this ever be supported? Love having 4 player on Gamecube would be great to have the same on PS2.

**Comment by darthcloud** [comment]
No this can't be done because their is not enough IO on the ESP32.

And PS2 multitap can't be emulated because it's too fast.

**Reply by Bruce285** [reply]
Shouldn't be a SPI frequency of 25 MHz be possible with esp32?

Hardware Multitap uses 1 controller slot and 1 memorycard slot - if instead of the second controller slot the first memorycard slot is connected to the esp32, could it be possible to connect 4 controllers via emulated multitap?

**Reply by icd2k3** [reply]
> And PS2 multitap can't be emulated because it's too fast.

aw damn, is this still true?

**Comment by kaduleite** [comment]
> No this can't be done because their is not enough IO on the ESP32.
> 

First, thanks for your great job. It's really amazing.

In  this case, is it possible to mount multitap and add two ESP32? So we could get 4 Bluetooth ports? Or that would cause Bluetooth conflict?

**Comment by konwektor** [comment]
possible to use port i/o expander?
like this one for example [https://www.learningaboutelectronics.com/Datasheets/MCP23017-IO-port-expander-datasheet.pdf](url)

**Comment by FatBaldDad** [comment]
I would like to try this. It seems like if you had two BR cores connectesd to a mulitap it would work.  Then you would need to figure out how to disconnect the multitap and redirect one of the cores back to the two console ports (esential by passing the multitap).  Might need to setup a third micro controller to manage the connections.
